environs/interface.go

```diff
@@ -60,6 +61,24 @@ type EnvironProvider interface {
 	PrepareConfig(context.Context, PrepareConfigParams) (*config.Config, error)
 }
 
+// ModelConfigProvider represents an interface that a [EnvironProvider] can
+// implement to provide opinions and defaults into a models config.
+type ModelConfigProvider interface {
+	// ConfigDefaults returns the default values for the
+	// provider specific config attributes.
+	ConfigDefaults() schema.Defaults
+
+	// ConfigSchema returns extra config attributes specific
+	// to this provider only.
+	ConfigSchema() schema.Fields
+
+	// ModelConfigDefaults provides a set of default model config attributes
+	// that
+	// should be set on a models config if they have not been specified by the
+	// user.
+	ModelConfigDefaults(context.Context) (map[string]any, error)
```

Comment:
question: why provide a context which is never used?

Reply:
Trying to future proof. Defaults could be calculated in a number of ways include http calls to public api's (possible for ec2) to gather information. Trying to get ahead of the curve now and wire it up.